lib/extensions/descendant_loader.rb

```diff
@@ -180,6 +186,7 @@ def save_cache!
       return unless @cache_dirty
       cache_path.parent.mkpath
       cache_path.open('w') do |f|
```

Comment:
NOTE: might need to open this "rw" to prevent truncation before obtaining the lock according to https://apidock.com/ruby/File/flock

Reply:
we also have a file ownership bug here
It shows up when a `root` process opens a file owned by user `manageiq`
the resulting file will be owned by `root` and cause a read error